frame/staking/src/lib.rs

```diff
@@ -2518,31 +2548,36 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// Basic and cheap checks that we perform in validate unsigned, and in the execution.
-	pub fn pre_dispatch_checks(score: PhragmenScore, era: EraIndex) -> Result<(), Error<T>> {
+	///
+	/// State reads: ElectionState, CurrentEr, QueuedScore.
+	///
+	/// This function does weight refund in case of errors, which is based upon the fact that it is
+	/// called at the very beginning of the call site's function.
```

Comment:
AFAICT the complete weight except for the DB costs is refunded in case of errors.
I'm not sure it is fair to assume that there are no computation costs associated at all.
Is this deliberate?

Reply:
Okay I am confused now: If we return an error the weight is refunded? but the fee is still charged? this doesn't make sense. In this case a dispatch that caused a lot of time to be wasted will mess up the block weight limits since it practically consumed no weight yet it took some time (assuming they still pay for it).

Reply:
In this function you return errors like `Error::<T>::PhragmenEarlySubmission.with_weight(T::DbWeight::get().reads(1))`. This is interpreted as "Error with the actual weight consumed", so there is a refund of `initial_weight - error_weight`.
I was just checking that you intentionally want to return an actual weight that consists only of database costs as you're doing below (e.g. `T::DbWeight::get().reads(1)`) and no computational costs.
I think this is probably fine because you have not executed any logic yet AFAICT, but just wanted to check.

Reply:
I think the fee is (paritally) refunded.

Reply:
I think the weight returned is correct because computational cost should be negligible when we do:
* `Self::era_election_status().is_open(),`
* `		if let Some(current_era) = Self::current_era() {`
* ```
	if let Some(queued_score) = Self::queued_score() {
			ensure!(
				is_score_better(queued_score, score),

Reply:
yeah so I ONLY refund in `pre_dispatch_checks` which are super cheap, trivial checks that if you fail, you were probably just not accurate or late. And you have not caused damage to the system, since these checks are super cheap. We literally do 3 reads and nothing more, and the refund only charges that.